operators/src/processing/expression/parser.rs

```diff
@@ -508,173 +528,215 @@ mod tests {
-                    f64::atan(a)
+                fn import_atan__1(a: Option<f64>) -> Option<f64> {
+                    a.map(f64::atan)
                 }
                 #[inline]
-                fn import_cos__1(a: f64) -> f64 {
-                    f64::cos(a)
+                fn import_cos__1(a: Option<f64>) -> Option<f64> {
+                    a.map(f64::cos)
                 }
                 #[inline]
-                fn import_sin__1(a: f64) -> f64 {
-                    f64::sin(a)
+                fn import_sin__1(a: Option<f64>) -> Option<f64> {
+                    a.map(f64::sin)
                 }
                 #[inline]
-                fn import_tan__1(a: f64) -> f64 {
-                    f64::tan(a)
+                fn import_tan__1(a: Option<f64>) -> Option<f64> {
+                    a.map(f64::tan)
                 }
+
+                #prelude
+
                 #[no_mangle]
-                pub extern "C" fn waves() -> f64 {
-                    import_cos__1(import_sin__1(import_tan__1(import_acos__1(import_asin__1(import_atan__1(1f64))))))
+                pub extern "Rust" fn waves() -> Option<f64> {
+                    import_cos__1(import_sin__1(import_tan__1(import_acos__1(import_asin__1(import_atan__1(Some(1f64)))))))
                 }
             }
             .to_string()
         );
 
         assert_eq!(
-            parse("non_linear", &[], &[], "ln(log10(pi()))"),
+            parse("non_linear", &[], "ln(log10(pi()))"),
             quote! {
                 #[inline]
-                fn import_ln__1(a: f64) -> f64 {
-                    f64::ln(a)
+                fn import_ln__1(a: Option<f64>) -> Option<f64> {
+                    a.map(f64::ln)
                 }
                 #[inline]
-                fn import_log10__1(a: f64) -> f64 {
-                    f64::log10(a)
+                fn import_log10__1(a: Option<f64>) -> Option<f64> {
+                    a.map(f64::log10)
                 }
                 #[inline]
-                fn import_pi__0() -> f64 {
-                    std::f64::consts::PI
+                fn import_pi__0() -> Option<f64> {
+                    Some(std::f64::consts::PI)
                 }
+
+                #prelude
+
                 #[no_mangle]
-                pub extern "C" fn non_linear() -> f64 {
+                pub extern "Rust" fn non_linear() -> Option<f64> {
                     import_ln__1(import_log10__1(import_pi__0()))
                 }
             }
             .to_string()
         );
 
         assert_eq!(
-            parse("three", &[], &[], "min(1, 2, max(3, 4, 5))"),
+            parse("three", &[],  "min(1, 2, max(3, 4, 5))"),
             quote! {
                 #[inline]
-                fn import_max__3(a: f64, b: f64, c: f64) -> f64 {
-                    f64::max(f64::max(a, b), c)
+                fn import_max__3(a: Option<f64>, b: Option<f64>, c: Option<f64>) -> Option<f64> {
+                    apply(
```

Comment:
don't we check twice for nodata now as this expands to two match blocks?

Reply:
yes

Reply:
[1bb1fd6](https://github.com/geo-engine/geoengine/pull/599/commits/1bb1fd66655bf6f9acdcc8d22459fa4650b42359)